srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditExistingRule.java

```diff
@@ -0,0 +1,77 @@
+package gov.va.med.srcalc.web.view.admin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import gov.va.med.srcalc.domain.model.Rule;
+import gov.va.med.srcalc.domain.model.ValueMatcher;
+import gov.va.med.srcalc.service.AdminService;
+import gov.va.med.srcalc.service.InvalidIdentifierException;
+
+/**
+ * <p>An interface for editing an existing {@link Rule}.</p>
+ * 
+ * <p>The high-level workflow is:</p>
+ * 
+ * <ol>
+ * <li>Construct an instance based on an existing rule.</li>
+ * <li>Present the user with the current values and allow her to update the
+ * properties as desired.</li>
+ * <li>Use the {@link EditRuleValidator} validator to validate the user's edits.</li>
+ * <li>Call {@link #applyToRule()} to update the target rule with the
+ * new properties.</li>
+ * </ol>
+ */
+public class EditExistingRule extends EditRule
+{
+    private final Rule fTarget;
+    private final AdminService fAdminService;
+
+    /**
+     * Constructs an instance to edit the given rule.
+     * @param target the rule to edit
+     * @param adminService to provide reference data (e.g.  to the user
+     */
+    public EditExistingRule(final AdminService adminService, final Rule target)
+    {
+        super(adminService, target);
+        fAdminService = adminService;
+        fTarget = target;
+    }
+    
+    /**
+     * Applies the specified changes to the target rule.
+     * @return the target rule for convenience
+     * @throws IllegalStateException if any changes are invalid
+     */
+    public Rule applyToRule() throws InvalidIdentifierException
+    {
+        // Set the necessary fields on fTarget
+        fTarget.setDisplayName(this.getDisplayName());
+        final List<ValueMatcher> matchers = new ArrayList<ValueMatcher>(getMatchers().size());
+        for(final ValueMatcherBuilder builder: getMatchers())
+        {
+            matchers.add(builder.buildNew(fAdminService));
+        }
+        fTarget.setMatchers(matchers);
+        fTarget.setRequired(this.isBypassEnabled());
+        fTarget.setSummandExpression(getSummandExpression());
+        return fTarget;
+    }
+    
+    /**
+     * Returns the rule that will have changes made to it.
+     */
+    public Rule getTargetRule()
+    {
+        return fTarget;
+    }
+    
+    @Override
+    public String toString()
+    {
+        return String.format("Name: %s Summand Expression: %s Matchers: %s Target Rule: %s", 
```

Comment:
You don't need to change this, but remember Guava's `MoreObjects.toStringHelper()`.